lib/Conversion/TorchToLinalg/Linear.cpp

```diff
@@ -448,6 +449,242 @@ class ConvertAtenLinearOp : public OpConversionPattern<AtenLinearOp> {
+                  ConversionPatternRewriter &rewriter) const override {
+    Location loc = op->getLoc();
+    MLIRContext *context = op->getContext();
+    Value grad = adaptor.grad_output();
+    Value input = adaptor.input();
+    Value weight = adaptor.weight();
+    Value groups = adaptor.groups();
+
+    Type elementType = grad.getType().cast<RankedTensorType>().getElementType();
+    if (!elementType.isa<mlir::FloatType>())
+      return op.emitError("unimplemented: non-floating point type");
+    size_t inRank = grad.getType().cast<RankedTensorType>().getRank();
+    if (inRank != 4)
+      return rewriter.notifyMatchFailure(
+          op, "unimplemented: only 2D convolution currently supported");
+
+    Type intType = IntegerType::get(context, 64);
+    auto castIndexToInt = [&](Value v) {
+      return rewriter.create<arith::IndexCastOp>(loc, intType, v);
+    };
+
+    SmallVector<int64_t> paddingInts;
+    if (!matchPattern(op.padding(), m_TorchConstantIntList(paddingInts))) {
+      return rewriter.notifyMatchFailure(
+          op, "only support constant padding values");
+    }
+    SmallVector<int64_t> strideInts;
+    if (!matchPattern(op.stride(), m_TorchConstantIntList(strideInts)))
+      return rewriter.notifyMatchFailure(op,
+                                         "only support constant int strides");
+    SmallVector<int64_t> dilationInts;
+    if (!matchPattern(op.dilation(), m_TorchConstantIntList(dilationInts)))
+      return rewriter.notifyMatchFailure(op,
+                                         "only support constant int dilations");
+
+    for (size_t i = 0; i < paddingInts.size(); i++) {
+      if (paddingInts[i] != 0)
+        return rewriter.notifyMatchFailure(op, "only support 0 padding");
+      paddingInts[i] += 1;
+    }
+
+    SmallVector<Value> paddingIntValues =
+        getAsConstantIntValues(rewriter, loc, paddingInts);
+    SmallVector<Value> dilationIntValues =
+        getAsConstantIntValues(rewriter, loc, dilationInts);
+    SmallVector<Value> strideIntValues =
+        getAsConstantIntValues(rewriter, loc, strideInts);
+
+    auto stridesAttr = rewriter.getI64VectorAttr(strideInts);
+    auto dilationAttr = rewriter.getI64VectorAttr(dilationInts);
+
+    Value c1 =
+        rewriter.create<arith::ConstantOp>(loc, IntegerAttr::get(intType, 1));
+    Value c0 =
+        rewriter.create<arith::ConstantOp>(loc, FloatAttr::get(elementType, 0));
+    Value ci0 =
+        rewriter.create<arith::ConstantOp>(loc, rewriter.getI64IntegerAttr(0));
+    Value groupEqual1 = rewriter.create<arith::CmpIOp>(
+        loc, arith::CmpIPredicate::eq, groups, c1);
+    rewriter.create<cf::AssertOp>(
+        loc, groupEqual1, rewriter.getStringAttr("expect groups to be 1"));
+
+    // Pad the input tensor according to padding.
+    SmallVector<int64_t> paddingIncludingNC = {0, 0};
+    paddingIncludingNC.insert(paddingIncludingNC.end(), paddingInts.begin(),
```

Comment:
nit: you can also use `append`:

https://llvm.org/doxygen/classllvm_1_1SmallVectorImpl.html#a0809c4018a9fd60e3149986d9f93c791